src/services/fluence-aurora-connector/modules/fluence_aurora_connector/src/jsonrpc/register_worker.rs

```diff
@@ -0,0 +1,290 @@
+};
+use crate::jsonrpc::request::RequestError;
+use crate::jsonrpc::{JsonRpcError, JsonRpcReq, JSON_RPC_VERSION};
+use crate::peer_id::serialize_peer_id;
+
+#[derive(Debug, Error)]
+pub enum RegisterWorkerError {
+    #[error("invalid worker id {1}: {0:?}")]
+    InvalidWorkerId(#[source] ParseError, &'static str),
+    #[error("error encoding function inputs: {0:?}")]
+    EncodeInput(#[from] ethabi::Error),
+    #[error("invalid deal contract addr: {0:?}")]
+    ParseDealAddr(#[source] clarity::Error),
+    #[error("invalid private key")]
+    InvalidPrivateKey(#[source] Box<dyn std::error::Error>),
+    #[error("error sending transaction to rpc: {0:?}")]
+    SendTransaction(#[from] RequestError),
+    #[error(transparent)]
+    JsonRpcError(#[from] JsonRpcError),
+}
+
+#[marine]
+pub fn register_worker(
+    pat_id: Vec<u8>,
+    worker_id: &str,
+    chain: ChainInfo,
+    deal_addr: String,
+) -> Vec<String> {
+    // get network id from rpc
+    // get gas price from rpc
+    // form tx
+    // sign
+    // send tx to rpc
+    let r: Result<_, RegisterWorkerError> = try {
+        let input = encode_call(pat_id, worker_id)?;
+        let nonce = load_nonce()?;
+        let gas_price = get_gas_price()?;
+        let endpoint = chain.api_endpoint.clone();
+        let tx = make_tx(input, chain, nonce, gas_price, deal_addr)?;
+        send_tx(tx, endpoint)?
+    };
+
+    match r {
+        Ok(_) => vec![],
+        Err(err) => vec![format!("{}", err)],
+    }
+}
+
+/// Send transaction to RPC
+fn send_tx(tx: String, api_endpoint: String) -> Result<String, RegisterWorkerError> {
+    let req = JsonRpcReq {
+        id: 0,
+        jsonrpc: JSON_RPC_VERSION.to_string(),
+        method: "eth_sendRawTransaction".to_string(),
+        params: vec![tx],
+    };
+    let response = send_jsonrpc(api_endpoint, req)?;
+    let tx_id: String = response.get_result()?;
+
+    Ok(tx_id)
+}
+
+/// Load nonce from KV
+fn load_nonce() -> Result<u128, RegisterWorkerError> {
+    Ok(0)
+}
+
+/// Increment nonce in KV
+fn increment_nonce(_nonce: u128) -> Result<(), RegisterWorkerError> {
+    Ok(())
+}
+
+/// Description of the `setWorker` function from the `chain.workers` smart contract on chain
+fn function() -> Function {
+    #[allow(deprecated)]
+    Function {
+        name: String::from("setWorker"),
+        inputs: vec![
+            Param {
+                name: String::from("patId"),
+                kind: ParamType::FixedBytes(32),
+                internal_type: None,
+            },
+            Param {
+                name: String::from("workerId"),
+                kind: ParamType::FixedBytes(32),
+                internal_type: None,
+            },
+        ],
+        outputs: vec![],
+        constant: None,
+        state_mutability: StateMutability::NonPayable,
+    }
+}
+
+/// Encode `setWorker` call to bytes
+fn encode_call(pat_id: Vec<u8>, worker_id: &str) -> Result<Vec<u8>, RegisterWorkerError> {
```

Comment:
But why? It's inside register_worker module, and it's private. Name has local meaning, and is not meant to be exposed.

Reply:
Because it's confusing to read, so since it's private and local we can easily give it some ugly but clear name

It'd be find if this thing was inside a trait though, outside of it, even being local, it breaks natural though process

I remember, Lyosha also wanted to rename smth because of it